platform/backend/repository/testbed/src/test/kotlin/io/hamal/repository/CodeTest.kt

```diff
@@ -450,29 +452,84 @@ internal class CodeRepositoryTest : AbstractUnitTest() {
 
     }
 
-    private fun CodeRepository.createCode(
-        codeId: CodeId,
-        groupId: GroupId,
-        cmdId: CmdId = CmdId(abs(Random(10).nextInt()) + 10),
-        codeValue: CodeValue
-    ) {
-        create(
-            CreateCmd(
-                id = cmdId,
-                codeId = codeId,
-                groupId = groupId,
-                value = codeValue
+    @Nested
+    inner class TimestampTest {
+
+        @TestFactory
+        fun `Initialization Test`() = runWith(CodeRepository::class) {
+            createCode(CodeId(1), GroupId(1), CmdGen(), CodeValue("1+1"))
+            Thread.sleep(1000)
+            update(
+                CodeId(1), UpdateCmd(
+                    id = CmdGen(),
+                    value = CodeValue("1 + 2")
+                )
             )
-        )
-    }
 
-    private fun CodeRepository.verifyCount(expected: Int) {
-        verifyCount(expected) { }
+            val t1 = get(CodeId(1), CodeVersion(1)).timestamp
+            val t2 = get(CodeId(1), CodeVersion(2)).timestamp
+            assertThat(t2, not(t1))
+        }
+
+        @TestFactory
+        fun `Ordering Test`() = runWith(CodeRepository::class) {
+            createCode(CodeId(1), GroupId(1), CmdGen(), CodeValue("1+1"))
+
+            val fn: (CodeVersion) -> RecordedAt = { get(CodeId(1), it).timestamp }
+            var last = fn(CodeVersion(1))
+            for (i in 1..5) {
+                Thread.sleep(1000) //This takes a while 1s is min
+                update(
+                    CodeId(1), UpdateCmd(
+                        id = CmdGen(),
+                        value = CodeValue("1 + ${i}")
+                    )
+                )
+                val next = fn(CodeVersion(i + 1))
+                assertThat(next, greaterThan(last))
+                last = next
+            }
+        }
+
+
     }
 
-    private fun CodeRepository.verifyCount(expected: Int, block: CodeQuery.() -> Unit) {
-        val counted = count(CodeQuery(groupIds = listOf()).also(block))
-        assertThat("number of code expected", counted, equalTo(expected.toULong()))
+
+    private object CmdGen {
+        private val atomicCounter = AtomicInteger(1)
+
+        operator fun invoke(): CmdId {
+            return CmdId(atomicCounter.incrementAndGet())
+        }
     }
 
```

Comment:
move this somehwere else so that this becomes reusable

Reply:
moved to abstract unit test, as in #78 , hope this doesnt conflict